File: Sully/Go/Sully.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

var xyz string

func main() {
	count := 5
	if (count <= 0) {
		return 
	}
	if (xyz == "FIRST") {
		count--
	}
	file := fmt.Sprintf("Sully_%d.go", count)
	s := "package main%c%cimport (%c%c%cfmt%c%c%c%cos%c%c%c%cos/exec%c%c)%c%cvar xyz string%c%cfunc main() {%c%ccount := %d%c%cif (count <= 0) {%c%c%creturn %c%c}%c%cif (xyz == %cFIRST%c) {%c%c%ccount--%c%c}%c%cfile := fmt.Sprintf(%cSully_%%d.go%c, count)%c%cs := %c%s%c%c%cs = fmt.Sprintf(s, 10, 10, 10, 9, 34, 34, 10, 9, 34, 34, 10, 9, 34, 34, 10, 10, 10, 10, 10, 10, 9, count, 10, 9, 10, 9, 9, 10, 9, 10, 9, 34, 34, 10, 9, 9, 10, 9, 10, 9, 34, 34, 10, 9, 34, s, 34, 10, 9, 10, 9, 10, 9, 10, 9, 34, 92, 34, 92, 34, 34, 10, 9, 10, 9, 34, 34, 34, 34, 10, 9, 10, 9, 9, 34, 34, 10, 9, 9, 10, 9, 9, 10, 9, 10, 10)%c%cb := []byte(s)%c%cos.WriteFile(file, b, 0644)%c%ccmd1 := %cgo build -ldflags %c%c-X main.xyz=FIRST%c%c -o Sully_%%d Sully_%%d.go%c%c%ccmd1 = fmt.Sprintf(cmd1, count, count)%c%cexec.Command(%cbash%c, %c-c%c, cmd1).Run()%c%cif count > 0 {%c%c%ccmd2 := %c./Sully_%%d%c%c%c%ccmd2 = fmt.Sprintf(cmd2, count)%c%c%cexec.Command(cmd2).Run()%c%c}%c}%c"
	s = fmt.Sprintf(s, 10, 10, 10, 9, 34, 34, 10, 9, 34, 34, 10, 9, 34, 34, 10, 10, 10, 10, 10, 10, 9, count, 10, 9, 10, 9, 9, 10, 9, 10, 9, 34, 34, 10, 9, 9, 10, 9, 10, 9, 34, 34, 10, 9, 34, s, 34, 10, 9, 10, 9, 10, 9, 10, 9, 34, 92, 34, 92, 34, 34, 10, 9, 10, 9, 34, 34, 34, 34, 10, 9, 10, 9, 9, 34, 34, 10, 9, 9, 10, 9, 9, 10, 9, 10, 10)
	b := []byte(s)
	os.WriteFile(file, b, 0644)
	cmd1 := "go build -ldflags \"-X main.xyz=FIRST\" -o Sully_%d Sully_%d.go"
	cmd1 = fmt.Sprintf(cmd1, count, count)
	exec.Command("bash", "-c", cmd1).Run()
	if count > 0 {
		cmd2 := "./Sully_%d"
		cmd2 = fmt.Sprintf(cmd2, count)
		exec.Command(cmd2).Run()
	}
}

File: Sully/C/Sully.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int count = 5;
char *s = "#include <stdio.h>%2$c#include <stdlib.h>%2$c#include <unistd.h>%2$c%2$cint count = %4$d;%2$cchar *s = %3$c%1$s%3$c;%2$c%2$cint main() {%2$c    count--;%2$c    char s1[100];%2$c    sprintf(s1, %3$cSully_%%d.c%3$c, count);%2$c    FILE *fptr = fopen(s1, %3$cw%3$c);%2$c    fprintf(fptr, s, s, 10, 34, count, 92);%2$c    fclose(fptr);%2$c    char cmd1[500];%2$c    sprintf(cmd1, %3$cbash -c %5$c%3$cgcc Sully_%%1$d.c -o Sully_%%1$d%5$c%3$c%3$c, count);%2$c    system(cmd1);%2$c    if (count) {%2$c        char cmd2[500];%2$c        sprintf(cmd2, %3$c./Sully_%%1$d%3$c, count);%2$c        system(cmd2);%2$c    }%2$c}%2$c";

int main() {
    count--;
    char s1[100];
    sprintf(s1, "Sully_%d.c", count);
    FILE *fptr = fopen(s1, "w");
    fprintf(fptr, s, s, 10, 34, count, 92);
    fclose(fptr);
    char cmd1[500];
    sprintf(cmd1, "bash -c \"gcc Sully_%1$d.c -o Sully_%1$d\"", count);
    system(cmd1);
    if (count) {
        char cmd2[500];
        sprintf(cmd2, "./Sully_%1$d", count);
        system(cmd2);
    }
}
